xds/src/main/java/io/grpc/xds/internal/sds/ServerWrapperForXds.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2020 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds.internal.sds;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import io.grpc.Server;
+import io.grpc.ServerServiceDefinition;
+import io.grpc.xds.XdsClientWrapperForServerSds;
+import java.io.IOException;
+import java.net.SocketAddress;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Wraps a {@link Server} delegate and {@link XdsClientWrapperForServerSds} and intercepts {@link
+ * Server#shutdown()} and {@link Server#start()} to shut down and start the
+ * {@link XdsClientWrapperForServerSds} object.
+ */
+final class ServerWrapperForXds extends Server {
+  private final Server delegate;
+  private final XdsClientWrapperForServerSds xdsClientWrapperForServerSds;
+
+  ServerWrapperForXds(Server delegate, XdsClientWrapperForServerSds xdsClientWrapperForServerSds) {
+    this.delegate = checkNotNull(delegate, "delegate");
+    this.xdsClientWrapperForServerSds =
+        checkNotNull(xdsClientWrapperForServerSds, "xdsClientWrapperForServerSds");
+  }
+
+  @Override
+  public Server start() throws IOException {
+    delegate.start();
+    xdsClientWrapperForServerSds.start();
```

Comment:
should those be in reverse order? since server is using the xdsClientWrapperForServerSds?

Reply:
`delegate.start()` can throw an `IOException` in which case we avoid starting the `xdsClientWrapperForServerSds` unnecessarily.  In terms of server depending on xdsClientWrapperForServerSds, having the reverse order doesn't guarantee anything: only when the xDS response arrives the server is ready to use the correct SslContext.

Reply:
Not changing the order based on my previous comment

Reply:
as discussed offline this raises concern about how we handle fallback in general. as the design is finalized, we will revisit.